shaft_element

```diff
@@ -0,0 +1,478 @@
+        position_l = [position, -self.o_d]  # lower
+        width = self.L
+        height = self.o_d - self.i_d
+
+        #  plot the upper half of the shaft
+        ax.add_patch(
+            mpatches.Rectangle(
+                position_u,
+                width,
+                height,
+                linestyle="--",
+                linewidth=0.5,
+                ec="k",
+                fc=self.color,
+                alpha=0.8,
+            )
+        )
+        #  plot the lower half of the shaft
+        ax.add_patch(
+            mpatches.Rectangle(
+                position_l,
+                width,
+                height,
+                linestyle="--",
+                linewidth=0.5,
+                ec="k",
+                fc=self.color,
+                alpha=0.8,
+            )
+        )
+
+    @classmethod
+    def section(
+        cls,
+        L,
+        ne,
+        si_d,
+        so_d,
+        material,
+        n=None,
+        shear_effects=True,
+        rotary_inertia=True,
+        gyroscopic=True,
+    ):
+        """Shaft section constructor.
+        This method will create a shaft section with length 'L'
+        divided into 'ne' elements.
+        Parameters
+        ----------
+        i_d : float
+            Inner diameter of the section.
+        o_d : float
+            Outer diameter of the section.
+        E : float
+            Young's modulus.
+        G_s : float
+            Shear modulus.
+        material : ross.material
+            Shaft material.
+        n : int, optional
+            Element number (coincident with it's first node).
+            If not given, it will be set when the rotor is assembled
+            according to the element's position in the list supplied to
+            the rotor constructor.
+        axial_force : float
+            Axial force.
+        torque : float
+            Torque.
+        shear_effects : bool
+            Determine if shear effects are taken into account.
+            Default is False.
+        rotary_inertia : bool
+            Determine if rotary_inertia effects are taken into account.
+            Default is False.
+        gyroscopic : bool
+            Determine if gyroscopic effects are taken into account.
+            Default is False.
+        Returns
+        -------
+        elements: list
+            List with the 'ne' shaft elements.
+        Examples
+        --------
+        >>> # shaft material
+        >>> from ross.materials import steel
+        >>> # shaft inner and outer diameters
+        >>> si_d = 0
+        >>> so_d = 0.01585
+        >>> sec = ShaftElement.section(247.65e-3, 4, 0, 15.8e-3, steel)
+        >>> len(sec)
+        4
+        >>> sec[0].i_d
+        0
+        """
+
+        le = L / ne
+
+        elements = [
+            cls(le, si_d, so_d, material, n, shear_effects, rotary_inertia, gyroscopic)
+            for _ in range(ne)
+        ]
+
+        return elements
+
+    @classmethod
```

Comment:
This `@classmethod` can be removed for now since we are going to implement io later.